routing/heap.go

```diff
@@ -53,13 +59,21 @@ func (d *distanceHeap) Less(i, j int) bool {
 // NOTE: This is part of the heap.Interface implementation.
 func (d *distanceHeap) Swap(i, j int) {
 	d.nodes[i], d.nodes[j] = d.nodes[j], d.nodes[i]
+	d.pubkeyIndices[d.nodes[i].node] = i
+	d.pubkeyIndices[d.nodes[j].node] = j
 }
```

Comment:
Would it be useful to add a basic test around this heap now that it does a bit more than what all the examples show?

Reply:
Sort of addressed, I don't think it's possible to compare `pubkeyIndices` maps between a mock heap (an array sorted by `sort.Slice`) and the actual heap.  Instead, I added a little code snippet that in `TestHeapOrdering` that validates that `pubkeyIndices` has the correct indices for `d.nodes`